src/accelerate/utils/imports.py

```diff
@@ -431,3 +431,16 @@ def is_xpu_available(check_device=False):
 
 def is_dvclive_available():
     return _is_package_available("dvclive")
+
+
+def is_torchdata_available():
+    return _is_package_available("torchdata")
+
+
+# TODO: Remove this function once stateful_dataloader is a stable feature in torchdata.
```

Comment:
Can this now be adjusted to use a version check?

Reply:
Yes, modeled after other version checks in file